python/foglamp/services/core/api/configuration.py

```diff
@@ -196,12 +196,13 @@
     config_item = urllib.parse.unquote(config_item) if config_item is not None else None
 
     data = await request.json()
-    # TODO: make it optimized and elegant
     cf_mgr = ConfigurationManager(connect.get_storage_async())
 
     try:
         value = data['value']
-        if not isinstance(value, str):
+        if isinstance(value, dict):
+            pass
```

Comment:
Bad data is still accepting for string type config items.

curl -sX PUT http://localhost:8081/foglamp/category/rest_api/authentication -d '{"value": {}}' | jq
{
  "default": "optional",
  "type": "string",
  "value": {},
  "description": "API Call Authentication (mandatory or optional)"
}


I believe for full proof interim solution for "string" type, we should add below code snippet in `foglamp.common.configuration_manager.ConfigurationManager#_validate_type_value`
```
        elif _type == 'string':
            if not isinstance(_value, str):
                return False
            return True
```

Reply:
for JSON we already have dict and json.loads based check